nimble/host/include/host/ble_audio_broadcast.h

```diff
@@ -0,0 +1,245 @@
+ */
+int ble_audio_broadcast_destroy(uint8_t adv_instance);
+
+/**
+ * @brief Update advertisements for given BASE configuration
+ *
+ * This function updates extended advertisements.
+ *
+ * @param[in] params            Pointer to structure with new advertising data
+ * @return                      0 on success;
+ *                              A non-zero value on failure.
+ */
+int ble_audio_broadcast_update(const struct ble_broadcast_update_params
+                               *params);
+
+/** BIG Subgroup parameters */
+struct ble_broadcast_subgroup_params {
+    /** Subgroup level Codec information */
+    struct ble_audio_codec_id *codec_id;
+
+    /** Subgroup level Codec Specific Configuration */
+    uint8_t *codec_spec_config;
+
+    /** Subgroup level Codec Specific Configuration length */
+    uint8_t codec_spec_config_len;
+
+    /** Subgroup Metadata */
+    uint8_t *metadata;
+
+    /** Subgroup Metadata length*/
+    uint8_t metadata_len;
+};
+
+/**
+ * @brief Build BIG subgroup structure
+ *
+ * This is a helper function can be used to fill out `ble_broadcast_subgroup`
+ * structure. Created subgroup extends subgroup list in provided BASE.
+ * This function increases `num_subgroups` in BASE structure.
+ *
+ * @param[out] base             Pointer to a `ble_audio_base` structure,
+ *                              that will be extended by the new subgroup.
+ *                              In case of error, filled out data may be
+ *                              erroneous.
+ * @param[out] subgroup         Pointer to a `ble_broadcast_subgroup`
+ *                              structure, that will be filled out with
+ *                              supplied configuration. In case of error,
+ *                              filled out data may be erroneous.
+ * @param[in] params            Pointer to a `ble_broadcast_subgroup_params`
+ *                              structure, containing information about new
+ *                              subgroup
+ *
+ * @return                      0 on success;
+ *                              A non-zero value on failure.
+ */
+int ble_audio_broadcast_build_sub(struct ble_audio_base *base,
+                                  struct ble_audio_big_subgroup *subgroup,
+                                  const struct ble_broadcast_subgroup_params
+                                  *params);
+
+/** BIS parameters */
+struct ble_broadcast_bis_params {
+    /** BIS index */
+    uint8_t idx;
+
+    /** BIS level Codec Specific Configuration */
+    uint8_t *codec_spec_config;
+
+    /** BIS level Codec Specific Configuration length */
+    uint8_t codec_spec_config_len;
+};
+
+/**
+ * @brief Build BIS structure
+ *
+ * This is a helper function can be used to fill out `ble_broadcast_bis`
+ * structure. Created BIS extends BIS list in provided subgroup.
+ * This function increases `bis_cnt` in subgroup structure.
+ *
+ * @param[out] sub              Pointer to a updated `ble_broadcast_subgroup`
```

Comment:
can we name it subgroup.

and also this is in/out param right?

Reply:
right, this should be `ble_audio_big_subgroup`. How to mark in/out params?